skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldYamlConfiguration.kt

```diff
@@ -16,25 +16,45 @@
 
 package com.google.container.tools.skaffold
 
+import com.intellij.openapi.project.Project
 import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiManager
 import org.yaml.snakeyaml.Yaml
 import java.io.ByteArrayInputStream
+import java.io.StringReader
 
 /**
  * Skaffold YAML configuration parsed into a map of objects. Useful objects and names, such
  * as [profiles], are additionally parsed and presented in a structured format.
  * Malformed YAML files are not handled and exception ([ScannerException]/[IOException]) is
  * thrown for the caller to handle.
+ *
+ * Optionally, can also take a [Project] parameter. If the project is not null, attempts to use
+ * project's PSI model to load current memory representation of the given Skaffold YAML.
  */
-class SkaffoldYamlConfiguration(skaffoldYamlFile: VirtualFile) {
+class SkaffoldYamlConfiguration(skaffoldYamlFile: VirtualFile, project: Project? = null) {
     /** Map of objects (presented as maps/lists) to their names from YAML file. */
     private val skaffoldYamlMap = mutableMapOf<Any, Any>()
 
     init {
+        // attempt to use in-memory data of PSI if exists and project is set first, or if not,
+        // fallback to directly parsing YAML file with VirtualFile
         val yamlLoader = Yaml()
-        skaffoldYamlMap.putAll(
-            yamlLoader.load(ByteArrayInputStream(skaffoldYamlFile.contentsToByteArray()))
-        )
+        var loadedWithPsiFile = false
+        if (project != null) {
```

Comment:
would the project ever be expected to be null? if not, could we just use this path?

Reply:
For now yes, to use the testing with Virtual Files, since setting up the same amount of testing with just PsiFile seems to be a large re-work since it always involve quite complex PsiManager (per project, not global service) and setting up at least some degree of PsiElements (root element with Document and maybe something else).

Reply:
I think you can test it by having the test file as an actual file in the resource directory: see https://github.com/GoogleCloudPlatform/google-cloud-intellij/blob/6929429dde951d404af995509d114b1d4265e58a/app-engine/java/testSrc/com/google/cloud/tools/intellij/appengine/java/project/DefaultAppEngineProjectServiceTest.java#L118

but lgtm for now

Reply:
Ah looks good, we can have this as a fix for exception, and I'll tackle change of Virtual Files to PSI Files next, might use the new test annotation for it as well.